dissect/ntfs/mft.py

```diff
@@ -122,10 +123,30 @@ def get(self, ref: Union[int, str, Instance], root: Optional[MftRecord] = None)
 
     def segments(self) -> Iterator[MftRecord]:
         """Yield all valid MFT records, regardless if they're allocated or not."""
```

Comment:
```suggestion
    def segments(self, start: int = 0, end: int = -1) -> Iterator[MftRecord]:
        """Yield all valid MFT records, regardless if they're allocated or not."""
        end = (self.get(FILE_NUMBER_MFT).size() // self.record_size) if end == -1 else end
```
I was thinking more something like this. Just a small modification to the existing `segments` method.

Reply:
Like this [`42b2942` (#29)](https://github.com/fox-it/dissect.ntfs/pull/29/commits/42b2942200f75ebeccd3cfc68a2abdbca2977242)?

Reply:
Yup, thanks!